src/net.cpp

```diff
@@ -845,6 +852,14 @@ static bool CompareNodeTXTime(const NodeEvictionCandidate &a, const NodeEviction
     return a.nTimeConnected > b.nTimeConnected;
 }
 
+// Pick out the potential block-relay only peers, and sort them by last block time.
```

Comment:
If you have to modify,  you might change to non-tx-relay peers/`CompareNonTxRelayPeers` as comment above the sort is referring to.